### PR TITLE
Add basic tests infrastructure

### PR DESCRIPTION
**Pull request by VictorVelarde**
- [x] Basic tooling with jest & react-testing-library
- [x] Tests on basic widget models (with a small refactor on them)
- [x] Tests on UI widgets for empty state

More tests are required, but they will be added in upcoming PRs

### CODE REVIEW

**Comment by VictorVelarde** [comment]
Some, still WIP, tests are commented, so you can merge this. Hopefully this will be useful for adding tests too on client-side calculations for widgets

File: src/tests/ui/widgets/CategoryWidgetUI.test.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import { CategoryWidgetUI } from 'src/ui';
```
Comment (AdriSolid) [inline src/tests/ui/widgets/CategoryWidgetUI.test.js:3]: We can add a `jsonconfig` file right? For relative paths from `src`

File: src/ui/widgets/FormulaWidgetUI.js [hunk 2]
```diff
@@ -60,14 +60,14 @@ function FormulaWidgetUI(props) {
     <Box className={classes.root}>
       {typeof formattedValue === 'object' && formattedValue !== null ? (
         <span>
-          <span className={`${classes.unit} ${classes.before}`}>
+          <span aria-label="preffix" className={`${classes.unit} ${classes.before}`}>
```
Comment (AdriSolid) [inline src/ui/widgets/FormulaWidgetUI.js:63]: "prefix" ? :roll_eyes:

File: src/ui/widgets/HistogramWidgetUI.js [hunk 3]
```diff
@@ -291,6 +293,18 @@ function HistogramWidgetUI(props) {
     click: clickEvent,
   };
 
+  if (data.length === 0) {
+    return (
+      <div>
+        <Alert severity='warning'>
+          <AlertTitle>No data available</AlertTitle>
+      There are no results from your data. If you are using filters, try tweaking them or zoom / pan the map to adjust the
```
Comment (AdriSolid) [inline src/ui/widgets/HistogramWidgetUI.js:301]: We can take a look to react-i18n.. Keeping labels in a json, and thinking about a possible internationalization of the app https://react.i18next.com/

File: src/tests/ui/widgets/CategoryWidgetUI.test.js (hunk 1 above)
Comment (albertoarana) [reply src/tests/ui/widgets/CategoryWidgetUI.test.js:3]: I agree

**Review by albertoarana — CHANGES_REQUESTED** [review]

**Comment by VictorVelarde** [comment]
This has been split into other smaller PRs. Can we close this now?

File: src/ui/widgets/FormulaWidgetUI.js (hunk 2 above)
Comment (VictorVelarde) [reply src/ui/widgets/FormulaWidgetUI.js:63]: Yeah, typo. Could you see if this has been fixed in your later PRs please?

File: src/ui/widgets/HistogramWidgetUI.js (hunk 3 above)
Comment (VictorVelarde) [reply src/ui/widgets/HistogramWidgetUI.js:301]: Aggree, but we decided to not enter into intl yet

File: src/ui/widgets/FormulaWidgetUI.js (hunk 2 above)
Comment (AdriSolid) [reply src/ui/widgets/FormulaWidgetUI.js:63]: It is not fixed!

**Comment by VictorVelarde** [comment]
I still need to review if any other detail here can be useful, before closing

**Comment by VictorVelarde** [comment]
Closing as work has been added in other PRs

**Pull request closed without merging**